tap_slack/streams.py

```diff
@@ -1,170 +1,641 @@
-from slack import WebClient
+import os
+from datetime import timedelta, datetime
 
+import pytz
 import singer
-import os
-import time
+from singer import metadata, utils
+from singer.utils import strptime_to_utc
 
-from singer import metadata
+from tap_slack.transform import transform_json
 
 LOGGER = singer.get_logger()
+DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S'
+utc = pytz.UTC
 
 
-class SlackStream():
+class SlackStream:
 
-    def __init__(self, webclient, config=None, catalog_stream=None, state=None):
-        self.webclient = webclient
+    def __init__(self, client, config=None, catalog=None, state=None, write_to_singer=True):
+        self.client = client
         self.config = config
-        self.catalog_stream = catalog_stream
+        self.catalog = catalog
         self.state = state
+        self.write_to_singer = write_to_singer
+        if config:
+            self.date_window_size = int(config.get('date_window_size', '7'))
```

Comment:
We mention briefly in the README about windows but not specifically this config and how it is used. Could we add some more detail to the README please, as it could be something that confuses new users without context.

Reply:
I'll add more info about this to the Readme.